### PR TITLE
Git intro

### PR DESCRIPTION
**Pull request by rarmatei**
Recommend clicking "View File" when looking at the lessons, as it's a **much** better experience:

image

### CODE REVIEW

File: introduction/week-2/lesson.md [hunk 1]
```diff
@@ -32,107 +32,148 @@ We will now do some pair programming.
...
 The group of people will be placed one each on numbered slots, so that your planners can refer to them by position.
 
-
 There will be one extra empty slot called ‘waiting-room’ (‘WR’ in the diagram).
 
-
 Your team can decide whether you will order the people in ascending or descending order.
 The Executor’s role
-* One of your team will take the role of executor during the test of your instructions
-* The executor will receive a command and the input needed to carry out the command (e.g. “COMPARE: is value in slot 3 is larger than value in slot 7?”)
```
Comment (rarmatei) [inline introduction/week-2/lesson.md]: Sorry for these! Just realised only after I committed that Prettier was running in the background and changed them.

Happy to revert them if you think it's worth it?

File: introduction/week-2/git-workshop-transcript.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+- you have lost your starting "template", that had a blank CV and cover letter. If you wanted to apply for a new job you would have to re-create the folder again and prepare both files again
+- if you make some changes to your CV, save them, turn off your computer, and the next day you realise you don't need them anymore, it will be really hard to undo them. You need to remember exactly what you changed, and only remove those changes.
+
+What we really want is to save occasional "[snapshots](https://www.merriam-webster.com/dictionary/snapshot)". As we're working through our job application, we want to take a "photo" of our folder each time we make an important change, so we can go back to it if we need to.
+
+So let's say our next task is to add a "Skills" section to our CV. We'll start by adding some skills, maybe decide to remove some, then add some more skills again, but once we're done and satisfied that we completed the task, we'll [commit](https://www.lexico.com/en/definition/commit) to being definitely done (which means we can "take a snapshot" of our folder).
```
Comment (bonboh) [inline introduction/week-2/git-workshop-transcript.md]: Minor nitpick: Would it be worth using the same online dictionary, e.g. changing this one (lexico) to merriam-webster for consistency?

File: introduction/week-2/git-workshop-transcript.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+<img alt="creating git folders" src="../assets/git/creating-folders.gif" width="400">
+
+As you can see, each time we commit to finishing a task, we create a new folder, give it the name of our task (so we can find it easier), and copy all our files in it.
+
+We now have these folders:
+
+<img alt="CV.txt file selected" src="../assets/git/3.png" width="250">
+
+The selected one, `job-application`, is our **latest** one. The one that has all the latest changes and is the most up to date. The other ones, contain different [milestones](https://www.merriam-webster.com/dictionary/milestone?src=search-dict-box) of our work, and they're named accordingly! There's a folder for our initial template (right before we first started writing) another one for when we added "Education" to our CV etc.
```
Comment (bonboh) [inline introduction/week-2/git-workshop-transcript.md]:
```suggestion
The selected one, `job-application`, is our **latest** one. It is the one that has all the latest changes and is the most up to date. The other ones, contain different [milestones](https://www.merriam-webster.com/dictionary/milestone) of our work, and they're named accordingly! There's a folder for our initial template (right before we first started writing) another one for when we added "Education" to our CV etc.
```

File: introduction/week-2/git-workshop-transcript.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+> Our second important word:
+> **add**
+
+---
+
+This is all really useful! But a bit annoying to do each time! And we're going to make a lot more changes - which means A LOT more folders. It will get confusing really fast.
+
+Would it not be nice if we had a robot we could ask to do all this work for us? We would just tell it anytime we're ready to commit, and it would create those folders for us, somewhere hidden, where we can't see them! Something like Alexa, or Siri - _"hey Alexa, add these files, I'm ready to commit them!"_
+
+Turns out we do, but our robot is called Git. And whenever we want to use Git in a folder, to track our changes, we need to tell it to initialize in that folder.
```
Comment (bonboh) [inline introduction/week-2/git-workshop-transcript.md]:
```suggestion
Turns out we do, but our robot is called Git. And whenever we want to use Git in a folder, to track our changes, we need to first tell it to initialize in that folder.
```

File: introduction/week-2/git-workshop-transcript.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+```
+git init
+git add CV.txt
+git commit -m "add skills"
+```
+
+Now it's your turn:
+
+1. Create a new folder with `mkdir`
```
Comment (bonboh) [inline introduction/week-2/git-workshop-transcript.md]:
```suggestion
1. Create a new folder with [`mkdir`](https://manpages.ubuntu.com/manpages/bionic/en/man1/mkdir.1posix.html)
```
Would it be useful to link to documentation? Maybe not if this Git section comes after the introduction to the terminal + commands

File: introduction/week-2/lesson.md (hunk 1 above)
Comment (bonboh) [reply introduction/week-2/lesson.md]: No problem, keep it prettied!

File: introduction/week-2/lesson.md [hunk 6]
```diff
@@ -32,107 +32,148 @@ We will now do some pair programming.
...
+- Discussion on functions. Why they could be useful?
 
 ## Git
 
-TBD
+👩‍🏫 Reading this in the class? Turn off your laptop screen and pay attention to the instructor. Ask questions!
+
+🏠 Reading this at home? Click [here](git-workshop-transcript.md) to access the full offline Git workshop.
+
+##### Workshop Exercises
```
Comment (bonboh) [inline introduction/week-2/lesson.md]:
```suggestion
### Workshop Exercises
```

File: introduction/week-2/lesson.md [hunk 7]
```diff
@@ -32,107 +32,148 @@ We will now do some pair programming.
...
+
+You'll find these in the full guide linked above as well.
+
+###### Part 1
```
Comment (bonboh) [inline introduction/week-2/lesson.md]:
```suggestion
#### Part 1
```

File: introduction/week-2/lesson.md [hunk 8]
```diff
@@ -32,107 +32,148 @@ We will now do some pair programming.
...
+5. `add` it to be committed
+6. `commit` it with a message
+7. create a cover letter file
+8. `add` it to be committed
+9. `commit` it with a message
+10. now make some changes to both files
+11. `add` both files to be committed
+12. `commit` your change with a message
+
+###### Part 2
```
Comment (bonboh) [inline introduction/week-2/lesson.md]:
```suggestion
#### Part 2
```

File: introduction/week-2/lesson.md [hunk 9]
```diff
@@ -32,107 +32,148 @@ We will now do some pair programming.
...
+8. `add` it to be part of the next commit
+9. now change the Cover Letter
+10. use `git status` (notice how one is green and one is red)
+11. `commit` with a message
+12. use `git status` (notice how there's only the red Cover Letter now)
+13. `add` the Cover Letter and `commit`
+14. use `git status` (notice how you have no files)
+15. use `git log` to see what changes you made
+
+###### Part 3
```
Comment (bonboh) [inline introduction/week-2/lesson.md]:
```suggestion
#### Part 3
```

**Review by bonboh — CHANGES_REQUESTED** [review]
Great stuff @rarmatei! Left just a few comments. I thought of Google Docs when reading this since it has [revision history](https://docs.google.com/document/d/1woFw1WgUn7-xyYMMOTnWjYxqWQ-fceXIzgefjk3dEkw)—would it be useful to mention this as well as another example of creating snapshots?

**Pull request merged**